modules/ROOT/pages/subqueries/call-subquery.adoc

```diff
@@ -0,0 +1,395 @@
+[source, cypher]
+----
+MATCH (person:Person)
+WITH person ORDER BY person.age ASC LIMIT 1
+  SET person:ListHead
+WITH *
+MATCH (next: Person)
+  WHERE NOT next:ListHead
+WITH next ORDER BY next.age
+CALL {
+  WITH next
+  MATCH (current:ListHead)
+    REMOVE current:ListHead
+    SET next:ListHead
+    CREATE(current)-[r:IS_YOUNGER_THAN]->(next)
+  RETURN current AS from, next AS to
+}
+RETURN
+  from.name AS name,
+  from.age AS age,
+  to.name AS closestOlderName,
+  to.age AS closestOlderAge
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="4*<m"]
+|===
+| +name+ | +age+ | +closestOlderName+ | +closestOlderAge+
+
+| "Alice" | 20 | "Bob" | 27
+| "Bob" | 27 | "Dora" | 30
+| "Dora" | 30 | "Charlie" | 65
+
+4+d|Rows: 3
+|===
+
+======
+
+
+[[call-post-union]]
+== Post-union processing
+
+Subqueries can be used to process the results of a `UNION` query further.
+This example query finds the youngest and the oldest person in the database and orders them by name.
+
+.Query
+[source, cypher]
+----
+CALL {
+  MATCH (p:Person)
+  RETURN p
+  ORDER BY p.age ASC
+  LIMIT 1
+UNION
+  MATCH (p:Person)
+  RETURN p
+  ORDER BY p.age DESC
+  LIMIT 1
+}
+RETURN p.name, p.age
+ORDER BY p.name
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| p.name | p.age
+| "Alice" | 20
+| "Charlie" | 65
+2+d|Rows: 2
+|===
+
+If different parts of a result should be matched differently, with some aggregation over the whole results, subqueries need to be used.
+This example query finds friends and/or parents for each person.
+Subsequently the number of friends and parents are counted together.
+
+.Query
+[source, cypher]
+----
+MATCH (p:Person)
+CALL {
+  WITH p
+  OPTIONAL MATCH (p)-[:FRIEND_OF]->(other:Person)
+  RETURN other
+UNION
+  WITH p
+  OPTIONAL MATCH (p)-[:CHILD_OF]->(other:Parent)
+  RETURN other
+}
+RETURN DISTINCT p.name, count(other)
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| p.name | count(other)
+| "Alice" | 2
+| "Bob" | 0
+| "Charlie" | 0
+| "Dora" | 0
+2+d|Rows: 4
+|===
+
+
+[[call-aggregation]]
+== Aggregations
+
+Returning subqueries change the number of results of the query: The result of the `CALL` clause is the combined result of evaluating the subquery for each input row.
+
+The following example finds the name of each person and the names of their friends:
+
+.Query
+[source, cypher]
+----
+MATCH (p:Person)
+CALL {
+  WITH p
+  MATCH (p)-[:FRIEND_OF]-(c:Person)
+  RETURN c.name AS friend
+}
+RETURN p.name, friend
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| p.name | friend
+| "Alice" | "Bob"
+| "Bob" | "Alice"
+2+d|Rows: 2
+|===
+
+The number of results of the subquery changed the number of results of the enclosing query: Instead of 4 rows, one for each node), there are now 2 rows which were found for Alice and Bob respectively. No rows are returned for Charlie and Dora since they have no friends in our example graph.
```

Comment:
Whilst the sentiment is true, I worry about the fact this query makes a simple query harder. This is just this query:

MATCH (p)-[:FRIEND_OF]-(c:Person)
RETURN p.name, c.name